Mathlib/Topology/Algebra/Module/LinearMap.lean

```diff
@@ -441,6 +441,28 @@ theorem comp_id (f : M₁ →SL[σ₁₂] M₂) : f.comp (id R₁ M₁) = f :=
 theorem id_comp (f : M₁ →SL[σ₁₂] M₂) : (id R₂ M₂).comp f = f :=
   ext fun _x => rfl
 
+section
+
+variable {R E F : Type*} [Semiring R]
+  [TopologicalSpace E] [AddCommMonoid E] [Module R E]
+  [TopologicalSpace F] [AddCommMonoid F] [Module R F]
+
+/-- `g ∘ f = id` as `ContinuousLinearMap`s implies `g ∘ f = id` as functions. -/
+lemma _root_.LeftInverse.of_composition {f : E →L[R] F} {g : F →L[R] E}
```

Comment:
These lemmas should have an easier proof, but I couldn't find it. If you give me a hint, I'll try again :-)
The part where I'm stuck is "coerce both sides of `hinv` to functions", then simp should do the trick.

Reply:
I [asked on zulip](https://leanprover.zulipchat.com/#narrow/channel/113489-new-members/topic/Apply.20a.20coercion.20to.20both.20sides.20of.20an.20equation/with/507889913)

Reply:
and golfed the proof accordingly